library/src/ui/index.js

```diff
@@ -1 +1,72 @@
-window.CoNDeT.ui = {}
\ No newline at end of file
+window.CoNDeT.ui = {
+    BaseComponent: {
+        state: {
+
+        },
+        common: {
+
+        },
+        init: function(common, props) {
+            this.ref = this.onInit();
+        },
+        update: function(common, props) {
+            children.update();
+            this.onUpdate();
+            this.ref.onUpdate();
+        },
+        destroy: function(common, props) {
+            children.destroy();
+            this.onDestroy();
+            this.ref.onDestroy();
+        },
+        getPosition: function() {
+            return {
+                x: this.domReference.offsetLeft,
+                y: this.domReference.offsetTop
+            };
+        },
+        getDimensions: function() {
+            return {
+                width: this.domReference.clientWidth,
+                height: this.domReference.clientHeight
+            }
+        },
+        containsPoint: function(x, y) {
+            return (x >= this.domReference.offsetLeft && 
+                x <= this.domReference.offsetLeft + this.domReference.clientWidth &&
+                y >= this.domReference.offsetTop &&
+                y <= this.domReference.offsetTop + this.domReference.clientHeight) ? true : false;
+        },
+        setState: function(state) {
+            this.state.onDestroy();
+            this.state = state;
+            this.state.onInit();
+        },
+        setupEventListeners: function() {
+            var self = this;
+            
+            ref.addEventListener('keyup', function() {
+                self.strategy.onKeyUp();
+            });
+            ref.addEventListener('keydown', function() {
+                self.strategy.onKeyDown();
+            });
+            ref.addEventListener('mouseup', function(component) {
+                component.onInit();
+                self.strategy.onMouseUp();
+            });
+            ref.addEventListener('mousedown', function(component) {
+                component.onInit();
+                self.strategy.onMouseDown();
+            });
+        },
```

Comment:
Przed `ref` brakuje `this` (znów odwołanie do global scope, co pokazuje dlaczego w JS jest to taki problem i dlaczego odchodzi się od takiego pisania bibliotek)
![chrome_xrocYLjrGe](https://user-images.githubusercontent.com/40248436/115952764-3ac09680-a4e8-11eb-8b0d-d460333473d4.png)